components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/AuthenticatorService.java

```diff
@@ -263,12 +263,20 @@ private String getApplicationScopes(String appName) throws APIManagementExceptio
         } else if (AuthenticatorConstants.ADMIN_APPLICATION.equals(appName)) {
             applicationRestAPI = RestApiUtil.getAdminRestAPIResource();
         }
+        //Todo: when all swaggers modified with no vendor extension, following swagger parser should be modified.
+        //todo: for now only publisher swagger have been modified for no vendor extensions
         try {
             if (applicationRestAPI != null) {
                 APIDefinition apiDefinitionFromSwagger20 = new APIDefinitionFromSwagger20();
-                Map<String, Scope> applicationScopesMap = null;
-                applicationScopesMap = apiDefinitionFromSwagger20.getScopes(applicationRestAPI);
-                scopes = String.join(" ", applicationScopesMap.keySet());
+                if (AuthenticatorConstants.PUBLISHER_APPLICATION.equals(appName)) {
```

Comment:
This will be fixed. Created an issue https://github.com/wso2/carbon-apimgt/issues/4414.